src/Compilers/CSharp/Test/Syntax/Parsing/CollectionLiteralParsingTests.cs

```diff
@@ -0,0 +1,3223 @@
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void TestError4()
+        {
+            UsingExpression("[..]",
+                // (1,4): error CS1525: Invalid expression term ']'
+                // [..]
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "]").WithArguments("]").WithLocation(1, 4));
+
+            N(SyntaxKind.CollectionCreationExpression);
+            {
+                N(SyntaxKind.OpenBracketToken);
+                N(SyntaxKind.SpreadElement);
+                {
+                    N(SyntaxKind.DotDotToken);
+                    M(SyntaxKind.IdentifierName);
+                    {
+                        M(SyntaxKind.IdentifierToken);
+                    }
+                }
+                N(SyntaxKind.CloseBracketToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void TestError5()
+        {
+            UsingExpression("[...e]",
+                // (1,2): error CS8635: Unexpected character sequence '...'
+                // [...e]
+                Diagnostic(ErrorCode.ERR_TripleDotNotAllowed, "").WithLocation(1, 2),
+                // (1,4): error CS1525: Invalid expression term '.'
+                // [...e]
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, ".").WithArguments(".").WithLocation(1, 4));
+
+            N(SyntaxKind.CollectionCreationExpression);
+            {
+                N(SyntaxKind.OpenBracketToken);
+                N(SyntaxKind.SpreadElement);
+                {
+                    N(SyntaxKind.DotDotToken);
+                    N(SyntaxKind.SimpleMemberAccessExpression);
+                    {
+                        M(SyntaxKind.IdentifierName);
+                        {
+                            M(SyntaxKind.IdentifierToken);
+                        }
+                        N(SyntaxKind.DotToken);
+                        N(SyntaxKind.IdentifierName);
+                        {
+                            N(SyntaxKind.IdentifierToken, "e");
+                        }
+                    }
+                }
+                N(SyntaxKind.CloseBracketToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void TestError6()
+        {
+            UsingExpression("[....]",
+                // (1,2): error CS8635: Unexpected character sequence '...'
+                // [....]
+                Diagnostic(ErrorCode.ERR_TripleDotNotAllowed, "").WithLocation(1, 2));
+
+            N(SyntaxKind.CollectionCreationExpression);
+            {
+                N(SyntaxKind.OpenBracketToken);
+                N(SyntaxKind.SpreadElement);
+                {
+                    N(SyntaxKind.DotDotToken);
+                    N(SyntaxKind.RangeExpression);
+                    {
+                        N(SyntaxKind.DotDotToken);
+                    }
+                }
+                N(SyntaxKind.CloseBracketToken);
+            }
+            EOF();
+        }
```

Comment:
Consider adding tests for:
* `A < B?[] > D` in statement, expression, `nameof`, and `typeof`/`default` contexts.
* `$"{[A:B]}"`
* `$"{[:]}"`
* `&[A,B][C]`
* `*[A,B]`
* `new [A, B]`